docs/API.md

```diff
@@ -74,6 +74,41 @@ Captures, also known as capturing groups, extract and store parts of the matched
 > [!NOTE]
 > TS Regex Builder does not have a construct for non-capturing groups. Such groups are implicitly added when required. E.g., `zeroOrMore(["abc"])` is encoded as `(?:abc)+`.
 
+### `backreference()`
+
+```ts
+function backreference(
+  groupNumber: number
+): Backreference
+```
+
+Regex syntax: `\1 \2`.
+
+A backreference is a way to match the same text as previously matched by a capturing group.
+
+### `namedCapture()`
```

Comment:
I've considered option of having a separate `namedCapture()` construct in addtion to basic `capture()`. After some prototyping and consulting I've found `capture(..., { name: 'aaa' })` to be better due to improving discoverability, and following JS convetion of "config" or "options" objects.